### ISSUE TITLE
Change LDAP password in web interface

### ISSUE THREAD

**Issue by consolidation** [ampache/ampache#808]
Hi everyone,

I'd like to ask about password changing and creating new users when using LDAP.
I have Ampache 3.6 configured to let user login with LDAP entries only (auth_methods="ldap")
This is working fine. But:
1) Is it possible to configure (or add php code somewhere in) Ampache so that an user can change his password under Preferences > Account > New Password, and this password will alo be updated in LDAP.
2) It is possible to create a new user in Ampache, and then this user will be added to LDAP automatically?

The reason I'm asking is that I'm not sure if these features are already implemented or not.

Thank you for your help

**Comment by Afterster** [comment]
This is not supported (even in current 3.8.0 development version), ldap is only used for authentication.
I believe to add this kind of feature changes would have to be made in /lib/class/user.class.php and /lib/class/auth.class.php.
If you have attempts to do the implementation like you suggested that would be great, it is not a bad feature idea and it could interest others :). Don't hesitate to suggest a PR in that case.

**Comment by Niols** [comment]
I just reworked the LDAP auth and put everything in a `LDAP` class.
I could write on this one. I've already written a profile editor for LDAP. However, for password, I only know how to change a password and replace it by a `SHA`-encrypted one. If someone with better knowledge could tell me how to support stronger encryptions (like `SSHA`), that would be great

**Comment by Niols** [comment]
Hello,

I'm considering implementing this feature.
However, I think I would need something to detect that a user belongs to an external backend.

So either we add a field in the MySQL base specifying from which external backend the user comes, or we do just like in auth: we try all backends in the specified order.

I believe the second one would be much easier to implement. @Afterster, would you have any idea, suggestions, …?

**Comment by Afterster** [comment]
I agree with you, the second one would be easier, based on `auth_methods` setting. Thanks again for taking care of this part @Niols
